include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -51,6 +51,10 @@ size_t getNumPorts(Operation *op);
 bool isConstant(Operation *op);
 bool isConstant(Value value);
 
+/// Returns true if this is a 'const' type that can only hold compile-time
+/// constant values
+bool isConst(Type type);
```

Comment:
Are there cases where you don't have a FIRRTLType which you would want to call this on?  Wouldn't all non-firrtltype be non-const anyway?

Reply:
I think adding `FIRRTLType::isConst()` would cover most cases. I recall a few instances where this would avoid casting from raw `Type`, but I can remove this for now.

Reply:
Oh, I am currently using it in the type printer logic, which uses `Type` instead of `FIRRTLType` for some reason

Reply:
Another example where this will come in handy is dealing with `firrtl::PortInfo`, which uses `Type`, not `FIRRTLBaseType` or `FIRRTLType`

Reply:
Yeah, that would explain why `Type` is used for ports. My point is that it's convenient for `isConst` to take `Type` as opposed to manually doing a dynamic cast to `FIRRTLBaseType` everywhere we have `Type` instead of `FIRRTLBaseType`. If you have a `FIRRTLBaseType`, directly calling `.isConst()` is preferable.